From 9f52ee340d303b236a042df272c2aa1c2f5946e0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Mon, 1 Jul 2024 11:26:07 +0200
Subject: [PATCH 1068/1695] dts: stm32: Fix ranges and dma-range property of
 mlahb node on stm32mp25

The parent node of mlahb is the root node which fixes size-cells to 2.
this implies that the bus address must de declared in 64-bit format.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 6036b2a768cb2d8cf29c116e4d7fcf0218d5172f dts: stm32: Fix ranges and dma-range property of mlahb node on stm32mp25]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I0abde2d18a31df0d7e020c83a548bd820e3c28a9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/396244
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 083115749..3e4847604 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -121,8 +121,8 @@
 		compatible = "st,mlahb", "simple-bus";
 		#address-cells = <1>;
 		#size-cells = <1>;
-		ranges = <0x0 0x0  0xFFFFFFFC>;
-		dma-ranges = <0x0 0x0 0xFFFFFFFC>;
+		ranges = <0x0 0x0 0x0 0xFFFFFFFC>;
+		dma-ranges = <0x0 0x0 0x0 0xFFFFFFFC>;
 
 		m33_rproc: m33@0 {
 			compatible = "st,stm32mp2-m33";
-- 
2.47.2

